src/main/java/com/thebois/models/beings/AbstractBeingGroup.java

```diff
@@ -8,50 +8,37 @@
  */
 public abstract class AbstractBeingGroup implements IBeingGroup {
 
-    private final Collection<IBeing> beings = new ArrayList<>();
+    private Collection<IBeing> beings = new ArrayList<>();
 
     /**
-     * Creates a BeingGroup containing 5 Beings.
-     */
-    public AbstractBeingGroup() {
-        final int numberOfBeings = 5;
-        createBeing(numberOfBeings);
-    }
-
-    /**
-     * Creates an AbstractBeingGroup containing n IBeings.
+     * Adds a being to the internal collection of beings.
      *
-     * @param numberOfBeings the number of IBeings to create.
+     * @param being IBeing to be added.
      */
-    public AbstractBeingGroup(int numberOfBeings) {
-        createBeing(numberOfBeings);
-    }
-
-    @Override
-    public Collection<IBeing> getBeings() {
-        return beings;
+    protected void addBeing(final IBeing being) {
+        beings.add(being);
     }
 
     @Override
     public void update() {
-        for (IBeing being : this.beings) {
+        for (final IBeing being : this.beings) {
             being.update();
         }
     }
 
-    /**
-     * Adds a being to the internal collection of beings.
-     *
-     * @param being IBeing to be added.
-     */
-    protected void addBeing(IBeing being) {
-        beings.add(being);
+    @Override
+    public Collection<IBeing> getBeings() {
+        final Collection<IBeing> clonedBeings = new ArrayList<>(beings.size());
+        clonedBeings.addAll(beings);
+        return clonedBeings;
     }
 
-    private void createBeing(int numberOfBeings) {
-        for (int i = 0; i < numberOfBeings; i++) {
-            addBeing(new Pawn());
-        }
+    protected void setBeings(final Collection<IBeing> beings) {
+        this.beings = beings;
+    }
+
+    protected Collection<IBeing> getBeingsByReference() {
```

Comment:
Do we need this?

Reply:
Not anymore, you're right! I had to create that method when the other `getBeings` returned deep clones of the real beings. `Colony` needs access to the actual `IBeing` instances, so I had to create this protected method. Forgot to remove it.